base/loading.jl

```diff
@@ -1020,13 +1024,14 @@ function require(uuidkey::PkgId)
         end
```

Comment:
seems suspect that we want to be holding a lock for this large of a chunk of (user) code, including the `include` and `package_callbacks` here, but we do need it for all of the shared state, which is quite a bit.